front/src/modules/trainschedule/components/Timetable/PacedTrain/hooks/useOccurrenceActions.ts

```diff
@@ -147,10 +149,68 @@ const useOccurrenceActions = ({
     [pacedTrain, occurrences, selectedTrainId]
   );
 
+  /**
+   * Resets an occurrence exceptions.
+   * If it is an indexed occurrence exception, it is totally removed.
+   * If it is an added exception, every change groups are removed except the start time.
+   */
+  const resetOccurrenceExceptions = useCallback(
+    (occurrenceId: OccurrenceId) => {
+      const occurrenceToUpdateException = findExceptionWithOccurrenceId(
+        pacedTrain.exceptions,
+        occurrenceId
+      );
+
+      if (!occurrenceToUpdateException) {
+        throw new Error('Cannot reset an occurrence which was not an exception');
+      }
+
+      let updatedExceptions = [...pacedTrain.exceptions];
+
+      if (isIndexedOccurrenceId(occurrenceId)) {
+        updatedExceptions = removeElementAtIndex(
+          updatedExceptions,
+          // If it is an indexed occurrence, the corresponding exception will always have an occurrence_index
+          occurrenceToUpdateException.occurrence_index!
+        );
+      } else {
+        // update occurrenceToUpdateException by removing all its properties except key and start time
+        const resetAddedException: PacedTrainException = {
+          key: occurrenceToUpdateException.key,
+          start_time: occurrenceToUpdateException.start_time,
+        };
+        const exceptionIndex = updatedExceptions.findIndex(
+          ({ key }) => resetAddedException.key === key
+        );
+        updatedExceptions = replaceElementAtIndex(
+          updatedExceptions,
+          exceptionIndex,
+          resetAddedException
+        );
+      }
```

Comment:
You can also simply use a map
```suggestion
        updatedExceptions = pacedTrain.exceptions.map((exception) =>
          exception.key === exceptionToUpdate.key
            ? {
                key: exceptionToUpdate.key,
                start_time: exceptionToUpdate.start_time,
              }
            : exception
        );
```

Reply:
Much better :) Done